app/controllers/active_sessions_controller.rb

```diff
@@ -2,16 +2,16 @@ class ActiveSessionsController < ApplicationController
   before_action :authenticate_user!
 
   def destroy
-    @active_session = current_user.active_sessions.find(params[:id])
+    active_session = current_user.active_sessions.find(params[:id])
 
-    @active_session.destroy
-
-    if current_user
-      redirect_to account_path, notice: "Session deleted."
-    else
+    if active_session == current_active_session
       forget_active_session
+      active_session.destroy
```

Comment:
Could we still call `active_session.destroy` before the `if` statement? I only ask because we're calling it in both branches.

Reply:
That bothered me as it doesn't feel DRY, but there's no good way to handle it. "current_active_session" will certainly be memoized by the time "destroy" is called because of "authenticate_user!".  But "current_active_session" has code to do a lookup for the comparison, and destroying it before then creates a dependency on "authenticate_user!" being called first.

So, yes, we can call it before the "if" statement, but I prefer everything to be explicit at the expense of an extra line.  This might save a headache later during refactoring.